packages/realm_dart/test/realm_object_test.dart

```diff
@@ -74,9 +74,390 @@ class _BoolValue {
+      subscription.cancel();
+    });
+
+    test('unknown keypath', () async {
+      var config = Configuration.local([Dog.schema, Person.schema]);
+      var realm = getRealm(config);
+
+      final dog = Dog("Mario");
+
+      realm.write(() {
+        realm.add(dog);
+      });
+
+      final externalChanges = <RealmObjectChanges<Dog>>[];
+      final subscription = dog.changesFor(["unknown"]).listen((changes) {
+        if (changes.properties.isNotEmpty) externalChanges.add(changes);
+      });
+
+      realm.write(() {
+        dog.age = 2;
+        dog.owner = Person("owner");
+      });
+
+      await verifyNotifications<Dog>(dog, externalChanges, ["age"]);
+      subscription.cancel();
+    });
+
+    test('property on links', () async {
+      var config = Configuration.local([TestNotificationObject.schema]);
+      var realm = getRealm(config);
+
+      final tno = TestNotificationObject();
+
+      realm.write(() {
+        realm.add(tno);
+      });
+
+      final externalChanges = <RealmObjectChanges<TestNotificationObject>>[];
+      final subscription = tno.changesFor(["link"]).listen((changes) {
+        if (changes.properties.isNotEmpty) externalChanges.add(changes);
+      });
+
+      // Property change
+      realm.write(() {
+        tno.link = TestNotificationObject();
+      });
+
+      await verifyNotifications<TestNotificationObject>(tno, externalChanges, ["link"]);
+
+      // Nested property change -- should raise
+      realm.write(() {
+        tno.link?.stringProperty = "NewVal";
+      });
+
+      await verifyNotifications<TestNotificationObject>(tno, externalChanges, null);
+
+      subscription.cancel();
+    });
+
+    test('nested property on links', () async {
+      var config = Configuration.local([TestNotificationObject.schema]);
+      var realm = getRealm(config);
+
+      final tno = TestNotificationObject();
+
+      realm.write(() {
+        realm.add(tno);
+      });
+
+      final externalChanges = <RealmObjectChanges<TestNotificationObject>>[];
+      final subscription = tno.changesFor(["link.stringProperty"]).listen((changes) {
+        if (changes.properties.isNotEmpty) externalChanges.add(changes);
+      });
+
+      // Top level property change
+      realm.write(() {
+        tno.link = TestNotificationObject();
+      });
+
+      await verifyNotifications<TestNotificationObject>(tno, externalChanges, ["link"]);
+
+      // Property on keypath changed -- should raise
+      realm.write(() {
+        tno.link?.stringProperty = "NewVal";
+      });
+
+      await verifyNotifications<TestNotificationObject>(tno, externalChanges, ["link"]);
+
+      // Property not on keypath changed -- should not raise
+      realm.write(() {
+        tno.link?.intProperty = 23;
+      });
+
+      await verifyNotifications<TestNotificationObject>(tno, externalChanges, null);
+
+      subscription.cancel();
+    });
+
+    // TODO Need to decide what we expect here....
+    test('mappedProperty', () async {
```

Comment:
This test at the moment passes, but we should decide what kind of behaviour we want to have with mapped properties. 
At the moment we need to pass the "managed" name when we subscribe, but we get the mapped name in `changes`